### ISSUE TITLE
Review and clean up tensorflow warnings

### ISSUE THREAD

**Issue by timpitman** [stellargraph/stellargraph#933]
### Description

We've got many warnings on CI from Tensorflow when running demo notebooks - while these are mostly stripped from the published outputs (except #826), Demos should ideally run without warnings.

We should go through and address as many warnings as possible (eg fix deprecated functions such as already done for #803 ) - in some cases raising new issues if there's significant work.

### Done Checklist

- [ ] Produced code for required functionality
- [ ] Tests written and coverage checked
- [ ] Code review performed
- [ ] Documentation on Google Docs  (if applicable)
- [ ] Documentation in repo
- [ ] Version number reflects new status
- [ ] CHANGELOG.md updated
- [ ] Team demo

**Comment by kieranricardo** [comment]
Warnings I've found:

- `ensemble-link-prediction-example`: `WARNING:tensorflow:Early stopping conditioned on metric `val_weighted_acc` which is not available. Available metrics are: loss,acc,val_loss,val_acc`
- `hinsage/movielens-recommender.ipynb`: `WARNING:tensorflow:multiprocessing can interact badly with TensorFlow, causing nondeterministic deadlocks. For high performance data pipelines tf.data is recommended



WARNING:tensorflow:multiprocessing can interact badly with TensorFlow, causing nondeterministic deadlocks. For high performance data pipelines tf.data is recommended.
 





WARNING:tensorflow:multiprocessing can interact badly with TensorFlow, causing nondeterministic deadlocks. For high performance data pipelines tf.data is recommended.
